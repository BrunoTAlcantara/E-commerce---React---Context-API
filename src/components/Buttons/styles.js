import styled, { css } from 'styled-components';
import { ShoppingCart as CartIcon } from '@styled-icons/material-outlined/ShoppingCart';

const buttonSize = {
  normal: () => css`
    width: 15rem;
    height: 4rem;
  `,
  big: () => css`
    width: 35rem;
    height: 4.2rem;
  `,
};
const buttonColor = {
  primary: (theme) => css`
    color: ${theme.colors.secondaryColor};
    background-color: ${theme.colors.primaryColor};
  `,
  secondary: (theme) => css`
    color: ${theme.colors.primaryColor};
    background-color: ${theme.colors.secondaryColor};
  `,
  disabled: (theme) => css`
    color: ${theme.colors.primaryColor};
    border: 0.3rem solid ${theme.colors.primaryColor};
    background: none;
  `,
};

export const Container = styled.button`
  ${({ theme, size, color }) => css`
    display: flex;
    align-items: center;
    justify-content: center;
    padding: 1rem;
    gap: 1rem;
    border: none;
    border-radius: 0.5rem;
    ${buttonSize[size]}
    ${buttonColor[color](theme)}
  `}
`;

export const Icon = styled(CartIcon)`
  width: 2.5rem;
  right: 2.5rem;
`;
