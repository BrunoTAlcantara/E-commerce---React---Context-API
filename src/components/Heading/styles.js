import styled, { css } from 'styled-components';

const titleSize = {
  small: (theme) =>
    css`
      font-size: ${theme.font.sizes.small};
    `,
  medium: (theme) =>
    css`
      font-size: ${theme.font.sizes.medium};
    `,
  big: (theme) =>
    css`
      font-size: ${theme.font.sizes.big};
    `,
  huge: (theme) =>
    css`
      font-size: ${theme.font.sizes.huge};
      ${mediaFont(theme)};
    `,
};
const titleWeight = {
  bold: (theme) =>
    css`
      font-weight: ${theme.font.weight.bold};
    `,
  eBold: (theme) =>
    css`
      font-weight: ${theme.font.weight.eBold};
    `,
  regular: (theme) =>
    css`
      font-weight: ${theme.font.weight.regular};
    `,
  thin: (theme) => css`
    font-weight: ${theme.font.weight.thin};
  `,
};
const titleCase = (uppercase) => css`
  text-transform: ${uppercase ? 'uppercase' : 'none'};
`;
const mediaFont = (theme) => css`
  @media ${theme.media.lteMedium} {
    font-size: ${theme.font.sizes.big};
  }
`;

export const Title = styled.h1`
  ${({ theme, colorDark, size, weight, uppercase }) => css`
    color: ${colorDark
      ? theme.colors.primaryColor
      : theme.colors.secondaryColor};
    ${titleCase(uppercase)}
    ${titleSize[size](theme)};
    ${titleWeight[weight](theme)};
  `}
`;
