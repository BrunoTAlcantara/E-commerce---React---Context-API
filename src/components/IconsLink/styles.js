import styled, { css } from 'styled-components';
import { AccountCircle } from '@styled-icons/material-outlined/AccountCircle';
import { ShoppingCart } from '@styled-icons/material-outlined/ShoppingCart';

export const Container = styled.div`
  ${({ theme }) => css``}
`;
export const Button = styled.button`
  ${({ theme }) => css`
    width: 3rem;
    height: 3rem;
    cursor: pointer;
    color: ${theme.colors.primaryColor};
    border: none;
    margin-left: 1rem;
    background: none;
    @media ${theme.media.lteMedium} {
      width: 2rem;
      height: 2rem;
      font-size: ${theme.font.sizes.medium};
    }
  `}
`;
