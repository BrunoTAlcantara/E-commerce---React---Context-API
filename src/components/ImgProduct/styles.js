import styled, { css } from 'styled-components';

const imgSize = {
  small: () => css`
    width: 8rem;
    height: 8rem;
  `,
  normal: (theme) => css`
    width: 30rem;
    height: 30rem;
    @media ${theme.media.lteMedium} {
      width: 20rem;
      height: 20rem;
    }
  `,
  big: (theme) => css`
    width: 56rem;
    height: 56rem;
    @media ${theme.media.lteMedium} {
      width: 30rem;
      height: 30rem;
    }
  `,
};

export const Container = styled.img`
  ${({ theme, size }) => css`
    ${imgSize[size](theme)}

    border-radius: 0.5rem;
    border: none;
  `}
`;
