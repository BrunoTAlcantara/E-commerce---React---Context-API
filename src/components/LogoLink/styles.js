import styled, { css } from 'styled-components';

export const Container = styled.a`
  ${({ theme }) => css`
    text-decoration: none;
    color: inherit;
    font-size: ${theme.font.sizes.big};
    font-weight: ${theme.font.weight.eBold};

    > img {
      max-height: 2.5rem;
    }
    @media ${theme.media.lteMedium} {
      max-height: 2.5rem;
      font-size: ${theme.font.sizes.medium};
    }
  `}
`;
