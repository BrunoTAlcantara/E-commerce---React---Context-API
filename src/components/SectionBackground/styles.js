import styled, { css } from 'styled-components';

export const Container = styled.div`
  ${({ theme }) => css`
    background-color: ${theme.colors.secondaryColor};
    color: ${theme.colors.primaryColor};
    min-height: 100vh;
    display: flex;
    align-items: center;
  `}
`;
