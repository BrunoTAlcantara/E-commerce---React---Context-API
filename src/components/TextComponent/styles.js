import styled, { css } from 'styled-components';

export const Container = styled.p`
  ${({ theme }) => css`
    font-size: ${theme.font.sizes.medium};
    font-weight: ${theme.font.weight.regular};
  `}
`;
