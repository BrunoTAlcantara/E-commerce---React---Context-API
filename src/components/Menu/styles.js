import styled, { css } from 'styled-components';
import { Container as SectionContainer } from '../SectionContainer/styles';

const menuVisible = (theme) => css`
  visibility: visible;
  opacity: 1;
`;

export const Container = styled.div`
  ${({ theme, visible }) => css`
    position: fixed;
    z-index: 5;
    top: 0;
    left: 0;
    right: 0;
    width: 100%;
    background: ${theme.colors.secondaryColor};
    border-bottom: #ddd;
    ${SectionContainer} {
      padding-top: 0;
      padding-bottom: 0;
    }
    @media ${theme.media.lteMedium} {
      visibility: hidden;
      opacity: 0;

      height: 100%;
      ${visible && menuVisible(theme)}

      > ${SectionContainer} {
        display: grid;
        grid-template-columns: 1fr;
        grid-template-rows: 1fr;
        height: 100vh;
        align-items: center;
        justify-content: center;
        overflow-y: auto;
      }
    }
  `}
`;
export const MenuContainer = styled.div`
  ${({ theme }) => css`
    display: flex;
    margin: 1rem;
    justify-content: space-between;
    align-items: center;
    @media ${theme.media.lteMedium} {
      display: block;
      text-align: left;
      padding: ${theme.font.spacing.big} 0;
    }
  `}
`;
export const Button = styled.button`
  ${({ theme }) => css`
    display: none;
    @media ${theme.media.lteMedium} {
      z-index: 6;
      display: block;
      position: fixed;
      top: 2rem;
      right: 2rem;
      width: 3rem;
      height: 3rem;
      background: ${theme.colors.secondaryColor};
      color: ${theme.colors.primaryColor};
      border: none;
    }
  `}
`;
