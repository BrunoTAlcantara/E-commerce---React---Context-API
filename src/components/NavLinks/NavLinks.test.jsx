import { screen } from '@testing-library/react';
import { renderTheme } from '../../styles/render-theme';
import { NavLinks } from '.';
import links from './mock';
import { theme } from '../../styles/theme';

describe('<NavLinks />', () => {
  it('should render links', () => {
    renderTheme(<NavLinks links={links} />);
    expect(screen.getAllByRole('link')).toHaveLength(links.length);
  });
  it('should not  render links', () => {
    renderTheme(<NavLinks />);
    expect(screen.queryAllByText(/links/i)).toHaveLength(0);
  });
  it('should render links', () => {
    renderTheme(<NavLinks links={links} />);
    expect(screen.getByText(/link2/i).parentElement).toHaveStyleRule(
      'flex-flow',
      'column wrap',
      {
        media: theme.media.lteMedium,
      },
    );
  });

  it('should match snapshot', () => {
    const { container } = renderTheme(<NavLinks links={links} />);
    expect(container.firstChild).toMatchSnapshot();
  });
});
