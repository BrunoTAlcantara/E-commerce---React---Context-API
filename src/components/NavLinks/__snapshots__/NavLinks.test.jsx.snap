// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavLinks /> should match snapshot 1`] = `
.c1 {
  display: block;
  -webkit-text-decoration: none;
  text-decoration: none;
  font-size: 1.6rem;
  font-weight: 700;
  color: #252A2C;
  padding: 0.8rem;
  position: relative;
}

.c1::after {
  content: '';
  position: absolute;
  bottom: 0;
  left: 50%;
  width: 0;
  height: 0.2rem;
  background: #252A2C;
  -webkit-transition: all 300ms ease-in-out;
  transition: all 300ms ease-in-out;
}

.c1:hover::after {
  left: 25%;
  width: 50%;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row wrap;
  -ms-flex-flow: row wrap;
  flex-flow: row wrap;
}

@media (max-width:768px) {
  .c0 {
    -webkit-flex-flow: column wrap;
    -ms-flex-flow: column wrap;
    flex-flow: column wrap;
    -webkit-align-content: center;
    -ms-flex-line-pack: center;
    align-content: center;
  }
}

<nav
  aria-label="Main Menu"
  class="c0"
>
  <a
    class="c1"
    href="#target1"
    target="_self"
  >
    link1
  </a>
  <a
    class="c1"
    href="#target2"
    target="_self"
  >
    link2
  </a>
  <a
    class="c1"
    href="#target3"
    target="_self"
  >
    link3
  </a>
  <a
    class="c1"
    href="#target4"
    target="_self"
  >
    link4
  </a>
</nav>
`;
